### PR TITLE
Refactor StatefulAuthProvider for static initialization and improved session management

### PR DESCRIPTION
**Pull request by pastuxso**
This refactor centralizes session management, reduces duplication, and improves maintainability across the extension.

- Centralized session change notifications using a new SessionChangeHandler implemented with RxJS and a singleton pattern.
- Refactored StatefulAuthProvider to follow a singleton pattern, simplifying its lifecycle and reducing dependencies.
- Added functionality to retrieve sessions from secrets storage, avoiding potential infinite loops when checking sessions inside a session change listener.
- Removed Kernel and RunmeUriHandler dependencies from StatefulAuthProvider for better modularity.
- Introduced a new GraphQL client instantiator to encapsulate implementation details like authentication token configuration.
- Ensured Webview refreshes correctly on login/logout events.

### CODE REVIEW

File: src/extension/extension.ts [hunk 1]
```diff
@@ -490,6 +490,14 @@ export class RunmeExtension {
 
     if (kernel.isFeatureOn(FeatureName.RequireStatefulAuth)) {
       await StatefulAuthProvider.instance.ensureSession()
+      StatefulAuthProvider.instance.currentSession().then(async (session) => {
```
Comment (sourishkrout) [inline src/extension/extension.ts:493]: @pastuxso don't we need to `await` this here?

Comment (pastuxso) [reply src/extension/extension.ts]: Let me know if you have further suggestions!

**Review by sourishkrout — APPROVED** [review]
✅ LGTM

**Comment by sourishkrout** [comment]
Let's break it in as a pre-release version.

**Pull request merged**